android/app/src/main/java/io/github/parkwithease/parkeasy/ui/login/LoginScreen.kt

```diff
@@ -71,81 +132,140 @@ fun LoginScreenInner(
+            events.onPasswordChange,
+            stringResource(R.string.password),
+            ImageVector.vectorResource(R.drawable.password),
+            KeyboardOptions(keyboardType = KeyboardType.Password),
+            visualTransformation = PasswordVisualTransformation(),
+        )
+        AnimatedVisibility(state.registering) {
+            LoginField(
+                state.confirmPassword,
+                events.onConfirmPasswordChange,
+                stringResource(R.string.confirm_password),
+                ImageVector.vectorResource(R.drawable.password),
+                KeyboardOptions(keyboardType = KeyboardType.Password),
+                isError = !state.matchingPasswords,
+                visualTransformation = PasswordVisualTransformation(),
+            )
+        }
+    }
 }
 
 @Composable
-fun PasswordField(text: String, onValueChange: (String) -> Unit, modifier: Modifier = Modifier) {
+private fun LoginField(
+    value: String,
+    onValueChange: (String) -> Unit,
+    label: String,
+    imageVector: ImageVector,
+    keyboardOptions: KeyboardOptions,
+    modifier: Modifier = Modifier,
+    isError: Boolean = false,
+    visualTransformation: VisualTransformation = VisualTransformation.None,
+) {
     OutlinedTextField(
-        value = text,
+        value = value,
         onValueChange = { onValueChange(it) },
-        label = { Text(stringResource(R.string.password)) },
-        leadingIcon = {
-            Icon(
-                imageVector = ImageVector.vectorResource(R.drawable.password),
-                contentDescription = stringResource(R.string.password_icon),
-            )
-        },
-        visualTransformation = PasswordVisualTransformation(),
-        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
+        label = { Text(label) },
+        leadingIcon = { Icon(imageVector = imageVector, contentDescription = label) },
+        isError = isError,
+        visualTransformation = visualTransformation,
+        keyboardOptions = keyboardOptions,
         singleLine = true,
         modifier = modifier,
     )
 }
 
 @Composable
-fun LoginButton(modifier: Modifier = Modifier, onClick: () -> Unit) {
-    Button(onClick = { onClick() }, modifier = modifier) { Text(stringResource(R.string.login)) }
+private fun LoginButton(label: String, onClick: () -> Unit, modifier: Modifier = Modifier) {
+    Button(onClick = { onClick() }, modifier = modifier) { Text(label) }
 }
 
 @Composable
-internal fun RegisterButton(modifier: Modifier = Modifier, onClick: () -> Unit) {
-    Button(onClick = { onClick() }, modifier = modifier, enabled = false) {
-        Text(stringResource(R.string.register))
+private fun ResetText(text: String, onClick: () -> Unit, modifier: Modifier = Modifier) {
+    Row(modifier, Arrangement.End) {
+        Text(text, Modifier.clickable { onClick() }, color = MaterialTheme.colorScheme.primary)
+    }
+}
+
+@Composable
+private fun SwitchText(
+    text: String,
```

Comment:
Can you come up with a better name for this field?
I am thinking of "labelText"